### PR TITLE
Caffe api doc

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?

Add Caffe API doc

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/1930

### CODE REVIEW

**Comment by wzhongyuan** [comment]
@jason-dai   does the naming and layout look good to you ?  Shall we move the doc into Model.md ? Thanks

**Comment by jason-dai** [comment]
We should make this a part of https://bigdl-project.github.io/master/#APIGuide/Module/

**Comment by wzhongyuan** [comment]
@jason-dai  ok, I will submit a revision shortly

**Comment by wzhongyuan** [comment]
@jason-dai  Shall we merge this PR? Thanks

**Comment by jason-dai** [comment]
You can merge it after it get reviewed by someone.

**Comment by wzhongyuan** [comment]
@SeaOfOcean   Please help review the doc, thanks

File: docs/docs/APIGuide/Module.md [hunk 1]
```diff
@@ -408,3 +408,89 @@ print "fc2 weight \n", fc2.element().parameters()['fc2']['weight']
...
+
+**Scala:**
+
+```scala
+Module.loadCaffeModel(defPath, modelPath)
+```
+**Python:**
+```python
+Model.load_caffe_model(defPath, modelPath)
+```
```
Comment (SeaOfOcean) [inline docs/docs/APIGuide/Module.md]: Add description to the parameter

File: docs/docs/APIGuide/Module.md [hunk 2]
```diff
@@ -408,3 +408,89 @@ print "fc2 weight \n", fc2.element().parameters()['fc2']['weight']
...
+**Scala:**
+
+```scala
+Module.loadCaffe(model, defPath, modelPath, match_all = true)
+```
+**Python:**
+```python
+Model.load_caffe(model, defPath, modelPath, match_all = True)
+```
+
```
Comment (SeaOfOcean) [inline docs/docs/APIGuide/Module.md]: explain paramter

File: docs/docs/APIGuide/Module.md [hunk 3]
```diff
@@ -408,3 +408,89 @@ print "fc2 weight \n", fc2.element().parameters()['fc2']['weight']
...
+**Scala:**
+
+```scala
+bigdlModel.saveCaffe(prototxtPath, modelPath, useV2 = true, overwrite = false)
+```
+**Python:**
+```python
+bigdl_model.save_caffe(prototxt_path, model_path, use_v2 = True, overwrite = False)
+```
+
```
Comment (SeaOfOcean) [inline docs/docs/APIGuide/Module.md]: explain parameter

Comment (SeaOfOcean) [reply docs/docs/APIGuide/Module.md]: note it only supports graph model

File: docs/docs/APIGuide/Module.md [hunk 4]
```diff
@@ -408,3 +408,89 @@ print "fc2 weight \n", fc2.element().parameters()['fc2']['weight']
...
+**Python example:**
+
+``` python
+from bigdl.nn.layer import *
+model = Model.load_caffe_model("/tmp/deploy.prototxt", "/tmp/caffe.caffemodel")
+```
+
+### Load weight from Caffe into pre-defined Model
```
Comment (SeaOfOcean) [inline docs/docs/APIGuide/Module.md]: Load weight from Caffe into pre-defined model (last model start with lower-case)

**Comment by wzhongyuan** [comment]
@SeaOfOcean  Please check refined one, thanks

File: docs/docs/APIGuide/Module.md [hunk 5]
```diff
@@ -448,6 +450,9 @@ Module.loadCaffe(model, defPath, modelPath, match_all = true)
 Model.load_caffe(model, defPath, modelPath, match_all = True)
 ```
 
+`model` is pre-defined BigDL model. Similar to `loadCaffemModule`, `defPath` and `modelPath` specify network deploy file and weight file,
+the forth parameter `math_all` specifies if layer definition should be exactly matched between pre-defined `mode` and the one from `defPath`
```
Comment (SeaOfOcean) [inline docs/docs/APIGuide/Module.md]: forth -> 4th or fourth

Comment (SeaOfOcean) [reply docs/docs/APIGuide/Module.md]: math_all -> match_all

**Comment by SeaOfOcean** [comment]
LGTM

**Pull request merged**